includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -307,6 +309,13 @@ public function update_intention(
 		if ( '' !== $selected_upe_payment_type ) {
 			// Only update the payment_method_types if we have a reference to the payment type the customer selected.
 			$request['payment_method_types'] = [ $selected_upe_payment_type ];
+
+			if (
+				CC_Payment_Method::PAYMENT_METHOD_STRIPE_ID === $selected_upe_payment_type &&
+				in_array( Link_Payment_Method::PAYMENT_METHOD_STRIPE_ID, \WC_Payments::get_gateway()->get_upe_enabled_payment_method_ids(), true )
+			) {
+				$request['payment_method_types'] = [ $selected_upe_payment_type, Link_Payment_Method::PAYMENT_METHOD_STRIPE_ID ];
```

Comment:
Do we override it for the stripe link specifically?

Reply:
By default, the payment intent is [updated](https://github.com/Automattic/woocommerce-payments/pull/4106/files/5759e7e7ed733e51a531eb15978f2f617dceb798#diff-0303646750b1a3d3223b823c3bc7345674ea5240a947c6ba00241ed4e182c41aR311) to support only 1 payment method -  'card' payment method in this case. 
If we have also the `link` payment method activated, we should allow also this payment method to be supported . So, here, I merged both payment method types ( card & link )

Reply:
:+1:, let's refactor this fragment towards better readability:

```php
			if (CC_Payment_Method::PAYMENT_METHOD_STRIPE_ID === $selected_upe_payment_type) {
				$is_link_enabled = in_array( Link_Payment_Method::PAYMENT_METHOD_STRIPE_ID, \WC_Payments::get_gateway()->get_upe_enabled_payment_method_ids(), true );
				if ($is_link_enabled) {
					$request['payment_method_types'] []= Link_Payment_Method::PAYMENT_METHOD_STRIPE_ID;
				}
			}
```